<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MW</class>
 <widget class="QMainWindow" name="MW">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>485</width>
    <height>898</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SMV2 buffered motion stream example</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Connection</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>SMV2 port device</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="portName">
           <property name="text">
            <string>FTDI0</string>
           </property>
           <property name="placeholderText">
            <string>Such as COM6 on Win or /dev/ttyUSB0 on Linux</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="connect">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="disconnect">
           <property name="text">
            <string>Disconnect</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Control</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QFrame" name="settingsGroup">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="1" column="1">
             <widget class="QSpinBox" name="setpointAmpl">
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>100000</number>
              </property>
              <property name="singleStep">
               <number>500</number>
              </property>
              <property name="value">
               <number>70000</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Number of axis to control</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="numOfAxis">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>16</number>
              </property>
              <property name="value">
               <number>4</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Sinusoidal setpoint frequency (Hz)</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="setpointFreq">
              <property name="minimum">
               <double>0.100000000000000</double>
              </property>
              <property name="maximum">
               <double>100.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
              <property name="value">
               <double>0.150000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Sinusoidal setpoint amplitude</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Setpoint sample rate (Hz)</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="samplerate">
              <item>
               <property name="text">
                <string>250</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>500</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>1000</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>1250</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2500</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Maximum buffer fill (%)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="maxBufferFillPercent">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>10.000000000000000</double>
           </property>
           <property name="value">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="startMotion">
           <property name="text">
            <string>Start motion</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="abortMotion">
           <property name="text">
            <string>Abort motion</string>
           </property>
           <property name="shortcut">
            <string>Esc</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clearFaults">
           <property name="text">
            <string>Clear faults</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string>Initial positions</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>DC</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="initX">
         <property name="minimum">
          <number>-100000</number>
         </property>
         <property name="maximum">
          <number>100000</number>
         </property>
         <property name="singleStep">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>STEPPER</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="initY">
         <property name="minimum">
          <number>-99999</number>
         </property>
         <property name="maximum">
          <number>99999</number>
         </property>
         <property name="singleStep">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>AC</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="initZ">
         <property name="minimum">
          <number>-99999</number>
         </property>
         <property name="maximum">
          <number>99999</number>
         </property>
         <property name="singleStep">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>LINEAR</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="initA">
         <property name="minimum">
          <number>-100000</number>
         </property>
         <property name="maximum">
          <number>100000</number>
         </property>
         <property name="singleStep">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="reset">
         <property name="text">
          <string>reset at init</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="setInitPositionsConstantly">
         <property name="text">
          <string>set initial positions constantly when not running sequence</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Buffer fill</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QProgressBar" name="bufferFill">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Log</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QTextEdit" name="log"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>portName</tabstop>
  <tabstop>connect</tabstop>
  <tabstop>disconnect</tabstop>
  <tabstop>startMotion</tabstop>
  <tabstop>abortMotion</tabstop>
  <tabstop>clearFaults</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
